templates/course-theme/locked-lesson-notice.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * The Template for displaying locked lesson notice in lesson page when using Course Theme.
+ *
+ * @author      Automattic
+ * @package     Sensei
+ * @category    Templates
+ * @version     3.15.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! function_exists( 'sensei_loked_lesson_notices_map' ) ) {
+	/**
+	 * Notices map to echo notices HTML.
+	 *
+	 * @param array $notice
+	 */
+	function sensei_loked_lesson_notices_map( $notice ) {
+		?>
+		<div class="sensei-lms-notice sensei-course-theme-locked-lesson-notice">
+			<?php if ( ! empty( $notice['title'] ) ) { ?>
+			<div class="sensei-course-theme-locked-lesson-notice__header">
+				<?php if ( ! empty( $notice['icon'] ) ) { ?>
+				<div class="sensei-course-theme-locked-lesson-notice__icon">
+					<?php
+					echo wp_kses(
+						Sensei()->assets->get_icon( $notice['icon'] ),
+						[
+							'svg'  => [
+								'width'   => true,
+								'height'  => true,
+								'viewBox' => true,
+								'xmlns'   => true,
+							],
+							'path' => [
+								'clip-rule' => true,
+								'fill-rule' => true,
+								'd'         => true,
+								'fill'      => true,
+							],
+						]
+					);
```

Comment:
I don't think that you need to sanitize the content that you're in control of. Better make sure the content is safe and does not need sanitizing in the first place.

However, if we think that `assets->get_icon( 'name' )` is not secure enough, then it's better to escape it's content inside the `get_icon` method. That way it already comes sanitized/escaped and we don't have to do it everywhere we use `get_icon` method.

Reply:
Hum! I didn't worry about that because we'll refactor the icons, but yeah! If we decide to go with this approach, it's better to already refactor it! =)

Good idea! Updated here: https://github.com/Automattic/sensei/pull/4509/commits/de205a25cc98bcaa03c0ffc9efa078e269b3b2db